### PR TITLE
[WIP][PHP] PHP 8.0 new syntax 

### LINKED ISSUE #991: [PHP] DNS_CAA type constant missing from PHP/PHP Source.sublime-syntax
```
echo DNS_CAA, DNS_A, DNS_CNAME, DNS_HINFO, DNS_MX, DNS_NS, 
     DNS_PTR, DNS_SOA, DNS_TXT, DNS_AAAA, DNS_SRV, DNS_NAPTR,
     DNS_A6, DNS_ALL, DNS_ANY;
```

If the above line is put into `PHP/syntax_test_php.php`, the first item, `DNS_CAA` is highlighted differently to the rest of the line. According to the [dns_get_record#type](http://php.net/manual/en/function.dns-get-record.php#refsect1-function.dns-get-record-parameters) parameter list, it is a valid constant.

### PR DESCRIPTION
**Pull request by jfcherng**
PHP 8.0 is scheduled to be feature-freezed on August 4 2020.

**References**:

- PHP 8.0 RFCs: https://wiki.php.net/rfc#php_80
- PHP 8.0 release schedule: https://wiki.php.net/todo/php80

### Need new parsing paths

- [X] [PHP RFC: Union Types 2.0](https://wiki.php.net/rfc/union_types_v2)
- [X] [PHP RFC: Attributes v2](https://wiki.php.net/rfc/attributes_v2) + [PHP RFC: Attribute Amendments](https://wiki.php.net/rfc/attribute_amendments) + [PHP RFC: Shorter Attribute Syntax](https://wiki.php.net/rfc/shorter_attribute_syntax)
- [X] [PHP RFC: Constructor Property Promotion](https://wiki.php.net/rfc/constructor_promotion)
- [X] [PHP RFC: Named Arguments](https://wiki.php.net/rfc/named_params)
- [ ] Maybe more...

### Just some simple supplements

- [x] [PHP RFC: Add Stringable interface](https://wiki.php.net/rfc/stringable)
- [X] [PHP RFC: Static return type](https://wiki.php.net/rfc/static_return_type)
- [X] [PHP RFC: Weak maps](https://wiki.php.net/rfc/weak_maps)
- [X] [PHP RFC: str_contains](https://wiki.php.net/rfc/str_contains)
- [X] [PHP RFC: get_debug_type](https://wiki.php.net/rfc/get_debug_type)
- [X] [PHP RFC: Object-based token_get_all() alternative](https://wiki.php.net/rfc/token_as_object)
- [X] [PHP RFC: Add str_starts_with() and str_ends_with() functions](https://wiki.php.net/rfc/add_str_starts_with_and_ends_with_functions)
- [X] [PHP RFC: Mixed Type v2](https://wiki.php.net/rfc/mixed_type_v2)
- [X] [PHP RFC: Match expression v2](https://wiki.php.net/rfc/match_expression_v2)
- [X] [PHP RFC: Nullsafe operator](https://wiki.php.net/rfc/nullsafe_operator)
- [X] [PHP PR: Add fdiv() function](https://github.com/php/php-src/pull/4769)
- [X] [PHP PR: Add get_resource_id() function](https://github.com/php/php-src/pull/5427)
- [X] [PHP PR: Introduce InternalIterator](https://github.com/php/php-src/pull/5216)
- [ ] Maybe more...

### Snippets 

- fn ...
- match ...

## Others

- Syntax definition uses `version: 2`
- Fixes #991

### CODE REVIEW

**Comment by wbond** [comment]
Based on your comment, this shouldn't really be reviewed until August, right?

**Comment by jfcherng** [comment]
> Based on your comment, this shouldn't really be reviewed until August, right?

Correct but I don't mind create another PR if you would like to merge this one now.

File: PHP/PHP Source.sublime-syntax [hunk 1]
```diff
@@ -547,10 +608,11 @@ contexts:
           pop: true
 
   class-body:
-    - meta_content_scope: meta.class.php meta.block.php
+    - meta_scope: meta.class.php meta.block.php
+    - include: attributes
     - include: comments
     - match: '\}'
-      scope: meta.class.php meta.block.php punctuation.section.block.php
+      scope: punctuation.section.block.php
```
Comment (wbond) [inline PHP/PHP Source.sublime-syntax:615]: I noticed this is missing the `.end` label. I searched the existing state of the syntax and there are a handful of other situations where `.begin` and `.end` are missing. Perhaps change them all?

```suggestion
      scope: punctuation.section.block.end.php
```

**Comment by wbond** [comment]
Awesome, thanks for all of these updates!

**Comment by willrowe** [comment]
@jfcherng I just noticed that named parameters are not working for function calls when they are closures stored in an array. Example:

```php
$array = [
    'callback' => function ($first, $second) {
        echo "{$first}, {$second}";
    },
];

$array['callback'](first: 'first', second: 'second');
```

The closing parentheses is not highlighted correctly:
Screenshot 2024-02-28 at 22 26 17

**Pull request merged**